assets/js/directives/settings-form.directive.js

```diff
@@ -12,12 +12,20 @@ function settingsForm () {
       scope.$parent.active = false;
       scope.$parent.reset();
 
+      scope.$parent.promptRecovery = () => {
+        scope.$parent.promptForRecovery = true;
+        scope.$parent.recoveryModal();
+      };
```

Comment:
This directive is meant to be an abstraction for every setting form. Could you leave this functionality in the manageSecondPassword controller?